lc-hax/Scripts/Modules/PossessionMod.cs

```diff
@@ -68,45 +83,132 @@ public sealed class PossessionMod : MonoBehaviour {
         Chat.Print($"Possess NoClip: {this.NoClipEnabled}");
 
         this.UpdateComponentsOnCurrentState(this.enabled);
     }
 
     // Updates movement components based on the current state
-    private void UpdateComponentsOnCurrentState(bool thisGameObjectIsEnabled) {
-        if (this.MousePan is not MousePan mousePan) {
-            return;
-        }
-
-        if (this.RigidbodyKeyboard is not RigidbodyMovement rigidbodyKeyboard) {
-            return;
-        }
-
-        if (this.Keyboard is not KeyboardMovement keyboard) {
-            return;
-        }
+    void UpdateComponentsOnCurrentState(bool thisGameObjectIsEnabled) {
+        if (this.MousePan is not MousePan mousePan) return;
+        if (this.RigidbodyKeyboard is not RigidbodyMovement rigidbodyKeyboard) return;
+        if (this.Keyboard is not KeyboardMovement keyboard) return;
 
         mousePan.enabled = thisGameObjectIsEnabled;
         rigidbodyKeyboard.enabled = !this.NoClipEnabled;
         keyboard.enabled = this.NoClipEnabled;
     }
 
     // Updates position and rotation of possessed enemy
-    void Update() {
-        _ = this.StartCoroutine(this.EndOfFrameCoroutine());
-    }
+    void Update() => _ = this.StartCoroutine(this.EndOfFrameCoroutine());
 
     // Coroutine for updating at end of frame
     IEnumerator EndOfFrameCoroutine() {
         yield return new WaitForEndOfFrame();
         this.EndOfFrameUpdate();
     }
 
+    void HandleEnemyMovements() {
+        if (this.EnemyToPossess is null) return;
+        switch (this.enemyIdentity) {
+            case EnemyIdentity.Nutcracker:
+                if (this.IsUsingPrimarySkill || this.IsUsingSecondarySkill) return; // prevent this from blocking the sentry skill of the nutcracker.
+                (this.EnemyToPossess as NutcrackerEnemyAI)?.OnMoving();
+                break;
+            case EnemyIdentity.Baboon:
+                break;
+            case EnemyIdentity.Centipede:
+                break;
+            case EnemyIdentity.Flowerman:
+                break;
+            case EnemyIdentity.ForestGiant:
+                break;
+            case EnemyIdentity.HoarderBug:
+                break;
+            case EnemyIdentity.Jester:
+                break;
+            case EnemyIdentity.Masked:
+                break;
+            case EnemyIdentity.MouthDog:
+                break;
+            case EnemyIdentity.Puffer:
+                break;
+            case EnemyIdentity.Sandworm:
+                break;
+            case EnemyIdentity.Springman:
+                break;
+            case EnemyIdentity.Blob:
+                break;
+            case EnemyIdentity.ElectricBees:
+                break;
+            case EnemyIdentity.Thumper:
+                break;
+            case EnemyIdentity.Default:
+                break;
+            case EnemyIdentity.None:
+                break;
+            default:
+                break;
+        }
+    }
```

Comment:
I have been trying to wonder what's the point of this? Do we just let the Nutcracker move on its own?